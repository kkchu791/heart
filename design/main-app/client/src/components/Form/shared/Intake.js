import React, {Component} from 'react'
import './Intake.scss'

const Intake = (WrappedComponent) => {
  console.log(WrappedComponent, "wrapped")
  return class extends Component {
    render() {
      return (
        <div className="intake">
          <div className="top-header"></div>
          <WrappedComponent />
        </div>
      )
    }
  }
}

export default Intake
