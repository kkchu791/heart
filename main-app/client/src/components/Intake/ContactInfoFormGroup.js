import React from 'react'
import { Field, ErrorMessage } from 'formik'
import { BasicField } from '../Form/shared'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(() => ({
  label: {
    fontSize: '16px',
    color: '#4f4f4f',
    display: 'block',
    paddingBottom: '4px',
    textAlign: 'left',
  },
  formInput: {
    padding: '12px 0',
  },
  starStyle: {
    color: 'red',
  },
}))

const form = [
  {
    component: BasicField,
    label: 'First Name',
    inputs: [
      {
        type: 'text',
        name: 'first_name',
        placeholder: 'First Name',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    label: 'Middle Name',
    inputs: [
      {
        type: 'text',
        name: 'middle_name',
        placeholder: 'Middle Name',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    label: 'Last Name',
    inputs: [
      { type: 'text', name: 'last_name', placeholder: 'Last Name', value: '' },
    ],
  },
  {
    component: BasicField,
    optional: true,
    label: 'Also Known As - AKA (Optional)',
    inputs: [
      {
        type: 'text',
        name: 'aka',
        placeholder: 'Also Known As - AKA (Optional)',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    label: "Driver's License / ID Number",
    inputs: [
      {
        type: 'text',
        name: 'dl',
        placeholder: "Driver's License / ID Number",
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    label: 'Date of Birth',
    inputs: [
      {
        type: 'date',
        name: 'dob',
        placeholder: 'Date of Birth',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    optional: true,
    label: 'Phone Number (optional)',
    inputs: [
      {
        type: 'tel',
        name: 'phone',
        placeholder: 'Phone Number (optional)',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    optional: true,
    label: 'Email Address (optional)',
    inputs: [
      {
        type: 'email',
        name: 'email',
        placeholder: 'Email Address (optional)',
        value: '',
      },
    ],
  },
  {
    component: BasicField,
    optional: true,
    label: 'Mailing Address (optional)',
    inputs: [
      {
        type: 'text',
        name: 'address',
        placeholder: 'Mailing Address (optional)',
        value: '',
      },
    ],
  },
]

const ContactInfoFormGroup = props => {
  console.log(props, 'props check it ')
  const classes = useStyles()
  return (
    <div className="contact-info-form-group">
      <h4>Contact Information</h4>

      {form.map(form_input =>
        renderInput(form_input, props.handleChange, props.handleBlur, props.values, classes, props.errors, props.touched)
      )}
    </div>
  )
}

const renderInput = (form_input, handleChange, handleBlur, values, classes, errors, touched) => {
  console.log(errors, "touched")
  return (
    <div key={form_input.label} className={classes.formInput}>
      <label className={classes.label}>
        {form_input.label}
        <span className={classes.starStyle}>{!form_input.optional && '*'}</span>
      </label>
      {form_input.inputs.map(input => {
        return (
          <div>
            <Field
              key={input.value}
              component={form_input.component}
              name={input.name}
              label={input.value}
              type={input.type}
              placeholder={input.placeholder}
              value={input.value}
              onChange={handleChange}
              values={values}
              error={errors[input.name] && touched[input.name]}
            />
          </div>
        )
      })}
    </div>
  )
}

export default ContactInfoFormGroup
