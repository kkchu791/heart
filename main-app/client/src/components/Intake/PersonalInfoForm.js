import React from 'react'
import { Formik, Form } from 'formik'
import { Button } from '@material-ui/core'
import ClinicFormGroup from './ClinicFormGroup'
import ContactInfoFormGroup from './ContactInfoFormGroup'
import GeneralInfoFormGroup from './GeneralInfoFormGroup'
import FamilyAndIncomeFormGroup from './FamilyAndIncomeFormGroup'
import AgreementsFormGroup from './AgreementsFormGroup'
import { Grid } from '@material-ui/core'
import { API_BASE_URL } from '../../config/url_config';
import { UserAuth } from '../../utilities/auth';
import axios from 'axios';

export const PersonalInfoForm = () => {

  const handleSubmit = (data) => {
    const authToken = UserAuth.getAuthToken();

    let config = {
      headers: {
        // Provide user's auth token as credentials
        Authorization: `Bearer ${authToken}`,
      }
    }
    const postData = async () => {
     const result = await axios.post(`${API_BASE_URL}/participants`, { data, timeout: 5000 }, config)
    };

    postData();
  }

  return (
    <Formik onSubmit={values => console.log(values, "onSubmit")}>
      {props => (
        <Form>
          <ClinicFormGroup {...props} />
          <ContactInfoFormGroup {...props} />
          <GeneralInfoFormGroup {...props} />
          <FamilyAndIncomeFormGroup {...props} />
          <AgreementsFormGroup {...props} />
          {/*The rest of the form groups could go here */}
          <br />
          <br />
          <Grid container justify="center" spacing={4}>
            <Grid item xs={6}>
              <Button
                variant="contained"
                size="large"
                color="default"
                onClick={() => handleSubmit(props.values)}
              >
                Save and Exit
              </Button>
            </Grid>
            <Grid item xs={6}>
              <Button
                variant="contained"
                size="large"
                color="primary"
              >
                Save and Continue
              </Button>
            </Grid>
          </Grid>
        </Form>
      )}
    </Formik>
  )
}
