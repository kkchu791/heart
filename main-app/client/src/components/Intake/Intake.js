import React from 'react'
import IntakeForm from './IntakeForm'
import { Formik, Form } from 'formik'
import { Button } from '@material-ui/core'
import FormTabs from '../Form/FormTabs'
import ProgramInfoFormGroup from './ProgramInfoFormGroup'
import OnsiteObligationsFormGroup from './OnsiteObligationsFormGroup'
import AgreementsFormGroup from './AgreementsFormGroup'
import {PersonalInfoForm} from './PersonalInfoForm'
import { Grid } from '@material-ui/core'

export const Intake = () => {
  const localStorageClick = data => {
    console.log(data)
    localStorage.setItem('data', JSON.stringify(data))
  }
  const localStorageDeleteClick = data => {
    console.log(data)
    localStorage.removeItem('data')
  }
  return (
    <div className="intake">
      {/*<div className="top-header" />*/}

      <IntakeForm>
        <FormTabs
          forms={[
            {
              label: 'Personal Information',
              Form: () => (
                <PersonalInfoForm />
              ),
            },
            {
              label: 'Obligations',
              Form: () => (
                <Formik
                  enableReinitialize={true}
                  initialValues={JSON.parse(localStorage.getItem('data'))}
                  onSubmit={values => console.log(values)}
                >
                  {({ handleReset, ...props }) => (
                    <Form>
                      <ProgramInfoFormGroup {...props} />
                      <OnsiteObligationsFormGroup {...props} />
                      {/*The rest of the form groups could go here */}
                      <br />
                      <br />
                      <Grid container justify="center" spacing={4}>
                        <Grid item xs={6}>
                          <Button
                            variant="contained"
                            size="large"
                            color="default"
                            onClick={() => {
                              localStorageClick(props.values)
                            }}
                          >
                            Save For Later
                          </Button>
                          <Button
                            variant="contained"
                            size="large"
                            color="default"
                            onClick={() => {
                              localStorageDeleteClick()
                            }}
                          >
                            Delete
                          </Button>
                        </Grid>
                        <Grid item xs={6}>
                          <Button
                            variant="contained"
                            size="large"
                            color="primary"
                          >
                            Continue to Agreement
                          </Button>
                        </Grid>
                      </Grid>
                    </Form>
                  )}
                </Formik>
              ),
            },
          ]}
        />
      </IntakeForm>
    </div>
  )
}
