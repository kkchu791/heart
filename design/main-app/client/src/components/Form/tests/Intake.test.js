import React from 'react';
import renderer from 'react-test-renderer';
import Intake from '../Intake.js';

it('renders correctly intake component', () => {
  const WrappedComponent = Intake(() => <div>I'm wrapped</div>);
  const IntakeComponent = renderer.create(<WrappedComponent />).toJSON();
  expect(IntakeComponent).toMatchSnapshot();
});
