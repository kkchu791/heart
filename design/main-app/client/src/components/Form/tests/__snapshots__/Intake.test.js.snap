// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly intake component 1`] = `
<div
  className="intake"
>
  <div
    className="top-header"
  />
  <div>
    I'm wrapped
  </div>
</div>
`;
